nlmod/plot/dcs.py

```diff
@@ -372,19 +379,73 @@ def plot_map_cs(
 
         return ax
 
-    def get_patches_array(self, z):
-        """Similar to plot_array function, only computes the array to update an existing plot_array.
+    def iterate_active_cells(self, zcs):
+        """Iterate over the cell indices of the cells in an array that are visible in the cross section and active in the model.
+        
+        Parameters
+        ----------
+        zcs : np.ndarray
+            2d array with dimensions (layer, distance along cross section).
+
+        Yields
+        ------
+        tuple
+            i, j indices of the cells that are active and visible in cross section. 
+        
+        """
+        for i in range(zcs.shape[0]):
+            for j in range(zcs.shape[1]):
+                if not (
+                    np.isnan(self.top[i, j])
+                    or np.isnan(self.bot[i, j])
+                    or np.isnan(zcs[i, j])
+                    or (self.bot[i, j] == self.zmax)
```

Comment:
Sorry got immediately distracted by your fancy animation so I checked out the code: should these be equality checks? or >=, <=?

Reply:
You are right